packages/calling/calling/src/CallingClient/CallingClient.ts

```diff
@@ -690,380 +451,16 @@ export class CallingClient extends Eventing<CallingClientEventTypes> implements
-      );
-    } else if (this.backupMobiusUris.length) {
-      log.log('Failing over to backup servers.', {
-        file: CALLING_CLIENT_FILE,
-        method: FAILOVER_UTIL,
-      });
-      abort = await this.registerDevice(FAILOVER_UTIL, this.backupMobiusUris);
-      if (!abort && !this.isDeviceRegistered()) {
-        interval = this.getRegRetryInterval();
-        setTimeout(async () => {
-          await this.mutex.runExclusive(async () => {
-            abort = await this.registerDevice(FAILOVER_UTIL, this.backupMobiusUris);
-            if (!abort && !this.isDeviceRegistered()) {
-              emitFinalFailure(this, FAILOVER_UTIL, CALLING_CLIENT_FILE);
-            }
-          });
-        }, interval * SEC_TO_MSEC_MFACTOR);
-        log.log(`Scheduled retry with backup servers in ${interval} seconds.`, {
-          file: CALLING_CLIENT_FILE,
-          method: FAILOVER_UTIL,
-        });
-      }
-    } else {
-      emitFinalFailure(this, FAILOVER_UTIL, CALLING_CLIENT_FILE);
-    }
-  }
-
-  /**
-   * Clears the failback timer if running.
-   */
-  private clearFailbackTimer() {
-    if (this.failbackTimer) {
-      clearTimeout(this.failbackTimer);
-      this.failbackTimer = undefined;
-    }
-  }
-
-  /**
-   * Returns true if device is registered with a backup mobius.
-   */
-  private failbackRequired(): boolean {
-    return this.isDeviceRegistered() && this.primaryMobiusUris.indexOf(this.url) === -1;
-  }
-
-  /**
-   * Calculates and returns a random value between rehomingIntervalMin and
-   * rehomingIntervalMax.
-   */
-  private getFailbackInterval(): number {
-    return Math.floor(
-      Math.random() * (this.rehomingIntervalMax - this.rehomingIntervalMin + 1) +
-        this.rehomingIntervalMin
-    );
-  }
-
-  /**
-   * Starts failback timer to move to primary mobius if device
-   * is registered with a backup mobius.
-   */
-  private initiateFailback() {
-    if (this.failbackRequired()) {
-      if (!this.failbackTimer) {
-        this.failback429RetryAttempts = 0;
-        const intervalInMinutes = this.getFailbackInterval();
-
-        this.startFailbackTimer(intervalInMinutes * MINUTES_TO_SEC_MFACTOR);
-      }
-    } else {
-      this.failback429RetryAttempts = 0;
-      this.clearFailbackTimer();
-    }
-  }
-
-  /**
-   * Starts failback timer with the interval value received.
-   *
-   * @param intervalInSeconds - Failback timer interval value in seconds.
-   */
-  private startFailbackTimer(intervalInSeconds: number) {
-    this.failbackTimer = setTimeout(
-      async () => this.executeFailback(),
-      intervalInSeconds * SEC_TO_MSEC_MFACTOR
-    );
-    log.log(`Failback scheduled after ${intervalInSeconds} seconds.`, {
-      file: CALLING_CLIENT_FILE,
-      method: this.startFailbackTimer.name,
-    });
-  }
-
-  /**
-   * Core logic for the failback processing, scheduled and executed
-   * at failback timer expiry.
-   */
-  private async executeFailback() {
+  public async register() {
```

Comment:
This change comes after the discussion we had to move all registration logic to keep it simple and encapsulated inside register.ts.

Reply:
I remember discussing about deregister() because we needed to call it from one of the methods in register.ts file, I still don't understand why did we need to change register API and as far as encapsulation is considered, that is already taken care because actual registration logic resides in registerDevice method only. Earlier it was register->registerDevice->createDevice but now it is register->triggerRegistration->attemptRegistrationForServers->postRegistration so how is this helping us in optimizing the code ?

Reply:
Earlier it was registerDevice->createDevice and registerDevice was like a god-method that has all the logic in callingClient.
In latest refactor,
triggerRegistration -> is the front-ending public api which calling client needs to call. 
attemptRegistrationWithServers -> is a method that is called many times internally by failover, failback logics and thus it needs a separate method. 
postRegistration -> deals with the REST request and was already existing, So didn't remove.

The methods are indeed more to segregate them based on their nature. More methods don't mean optimisation, but aids readability and reusability for sure.

Reply:
@BhargavSatya @Kesari3008 let's combine triggerRegistration and attemptRegistrationForServers because there isn't much difference between attemptRegistrationForServers and registerDevice. Both functions look equally big.